pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -837,8 +843,6 @@ func TestPodFailedSchedulingMultipleTimesDoesNotBlockNewerPod(t *testing.T) {
 
 	// Put in the unschedulable queue
 	q.AddUnschedulableIfNotPresent(newPodInfoNoTimestamp(&unschedulablePod), q.SchedulingCycle())
-	// Clear its backoff to simulate backoff its expiration
-	q.clearPodBackoff(&unschedulablePod)
```

Comment:
shouldn't we use a fake clock here too? Otherwise, how are we guaranteeing that this test is not flaky?

Reply:
yes we need, done.